### PR TITLE
Kafka performance metrics blog post.

### PR DESCRIPTION
**Pull request by badri**
TODO:
- [x] paste screenshots
- [x] articulate the testing loop
- [x] new relic
- [x] datadog
- [x] prom-grafana setup
- [ ] pod topologySpreadConstraints
- [ ] consumer properties

### CODE REVIEW

**Review by p6 — APPROVED** [review]

**Review by p6-platformatory — COMMENTED** [review]
@avinash-platformatory - can you look at this and sign off?
also address any pending gaps

File: _posts/2023-03-30-kafka-client-performance-metrics.md [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+3. As a sub-idea of 2, we don’t have an easy way to visualize these numbers. We all love to see the big picture instead of staring at a spreadsheet full of numbers. It would be great if we can pipe the output straight to tools like Grafana.
+
+## Apache Trogdor
+
+[Apache Trogdor](https://github.com/apache/kafka/blob/trunk/TROGDOR.md) is a test framework for Kafka. Although it does excellent performance testing, Trogdor emphasizes a lot of [fault testing](https://cwiki.apache.org/confluence/display/KAFKA/Fault+Injection). The idea is to run a Trogdor agent in every cluster node and a Trogdor coordinator daemon which manages the agents and tasks in the cluster. The agent can simulate a fault by crashing one of the brokers. We measure the message durability, adjust the parameters and re-run the process again. Confluent cloud [uses Trogdor](https://www.confluent.io/blog/cloud-kafka-as-a-service/) to performance test its infrastructure.
```
Comment (avinash-platformatory) [inline _posts/2023-03-30-kafka-client-performance-metrics.md]: Broken link for Apache Trogdor. Latest link - https://github.com/apache/kafka/tree/trunk/trogdor

File: _posts/2023-03-30-kafka-client-performance-metrics.md [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+
+Again, we found the following things lacking in Trogdor.
+
+1. No way to measure end to end latency. The [RoundTripWorkload](https://github.com/apache/kafka/blob/trunk/TROGDOR.md#roundtripworkload) is a close match for what we were looking for, but doesn’t quite cut it.
```
Comment (avinash-platformatory) [inline _posts/2023-03-30-kafka-client-performance-metrics.md]: Broken link for Apache Trogdor. Latest link - https://github.com/apache/kafka/tree/trunk/trogdor#roundtripworkload

Comment (avinash-platformatory) [inline _posts/2023-03-30-kafka-client-performance-metrics.md]: Reason why RoundTripWorkload does not fit our goal is not clear. From the docs, it seems like it produces and consumes the same message, allowing to measure the e2e latency.

File: _posts/2023-03-30-kafka-client-performance-metrics.md [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+  --if-not-exists \
+  --topic mytopic  \ # <-------- (2)
+  --create \
+  --bootstrap-server xxx-yyyy.us-west4.gcp.confluent.cloud:9092 \ # <-------- (3)
+  --replication-factor 3  \ # <-------- (4)
+  --partitions 1 \ # <-------- (5)
+  --command-config kafka.properties  # <-------- (6)
+```
+
+1. `kafka-topics` is the CLI tool which ships with your [confluent platform package](https://docs.confluent.io/platform/current/installation/installing_cp/zip-tar.html#prod-kafka-cli-install).
```
Comment (avinash-platformatory) [inline _posts/2023-03-30-kafka-client-performance-metrics.md]: Probably worth mentioning that kafka-topics.sh ships with Apache Kafka

File: _posts/2023-03-30-kafka-client-performance-metrics.md [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+As a quick example, if you want to use New Relic to process the metrics, your configuration would look like this:
+
+```yaml
+prometheus:
+  remote_write:
+  - url: https://metric-api.newrelic.com/prometheus/v1/write?prometheus_server=kafka-perf-test
+    bearer_token: xxxxxxxxxxxxxxxxxxxxxxxxxxx
+```
+
+The Helm charts and the associated code mentioned in this post can be found [here](https://github.com/Platformatory/confluent-performance-suite).
```
Comment (avinash-platformatory) [inline _posts/2023-03-30-kafka-client-performance-metrics.md:517]: The README for the repo is outdated and will be confusing for readers. There's mentions of S3, etc. Needs to be updated.

**Review by avinash-platformatory — CHANGES_REQUESTED** [review]

File: _posts/2023-03-30-kafka-client-performance-metrics.md (hunk 4 above)
Comment (p6-platformatory) [reply _posts/2023-03-30-kafka-client-performance-metrics.md:517]: Please also rename the repository. It has nothing to do with confluent.

**Comment by badri** [comment]
Raising a new PR as this is more than a yr old.

**Pull request closed without merging**